assembly/src/ast/imports.rs

```diff
@@ -0,0 +1,156 @@
+use super::{
+    BTreeMap, ByteReader, ByteWriter, Deserializable, DeserializationError, LibraryPath,
+    ParsingError, ProcedureId, ProcedureName, Serializable, String, ToString, Token, TokenStream,
+    Vec, MAX_IMPORTS, MAX_INVOKED_IMPORTED_PROCS,
+};
+
+// TYPE ALIASES
+// ================================================================================================
+
+type ImportedModulesMap = BTreeMap<String, LibraryPath>;
+type InvokedProcsMap = BTreeMap<ProcedureId, (ProcedureName, LibraryPath)>;
+
+// MODULE IMPORTS
+// ================================================================================================
+
+/// Information about imports stored in the AST
+#[derive(Debug, Default, Clone, PartialEq, Eq)]
+pub struct ModuleImports {
+    /// Imported libraries.
+    imports: ImportedModulesMap,
+    /// Imported procedures that are called from somewhere in the AST.
+    invoked_procs: InvokedProcsMap,
+}
+
+impl ModuleImports {
+    // CONSTRUCTOR
+    // --------------------------------------------------------------------------------------------
+    /// Create a new ModuleImports instance
```

Comment:
We should add `# Panics` section to the doc comment to document under which conditions this function can panic (there are a few examples of how we usually do it).